apps/src/applab/GoogleChart.js

```diff
@@ -105,17 +105,27 @@ export default class GoogleChart {
    * Makes sure data looks okay, throws errors and logs warnings as appropriate.
    * @param {string[]} columns
    * @param {Object[]} data
+   * @param {number} [minColumns = 2]
+   * @param {number} [maxColumns]
    * @private
    */
-  verifyData_(data, columns) {
+  verifyData_(data, columns, minColumns = 2, maxColumns) {
     // Warn when no rows are present
     if (data.length === 0) {
       this.warn('No data.');
     }
 
+    if (maxColumns && columns.length > maxColumns) {
+      this.warn(
+        `Too many columns for pie chart; only using the first ${maxColumns}.`
```

Comment:
It seems like a mistake to reference "pie chart" specifically in this base class.  It might be fine to make this message generic.